src/main/java/reposense/parser/ArgsParser.java

```diff
@@ -290,6 +290,10 @@ public static CliArguments parse(String[] args) throws HelpScreenException, Pars
 
             if (isSinceDateProvided) {
                 sinceDate = TimeUtil.getSinceDate(cliSinceDate.get());
+                // For --since d1, need to adjust the arbitrary date based on timezone
+                if (sinceDate.equals(SinceDateArgumentType.getArbitraryFirstCommitDateUtcLocal())) {
```

Comment:
Could this logic go into TimeUtil::getSinceDate?

Reply:
Would just like to get this clarified too - there seems to be an identical clause in `getSinceDate`. If it can't/shouldn't be moved into `getSinceDate`, I'll approve the PR.

Reply:
Now that you've mentioned it, I think the clause in `TimeUtil.getSinceDate` can be safely removed. Since the new arbitrary date is `1970-01-01 00:00:00` in UTC time, adding `withHour(0).withMinute(0).withSecond(0)` does nothing to it so it is effectively returning the same date. The equality check in `ArgsParser` should still pass.

That being said, we still have to keep `withHour(0).withMinute(0).withSecond(0)` in `TimeUtil.getSinceDate` since the `getDateMinusAMonth` and `getDateMinusNDays` methods still need to convert the since date to midnight.